Test/Core/Components/CollisionListener.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace Duality.Tests.Components
+{
+	/// <summary>
+	/// A collision listener to expose the collision events to tests
+	/// </summary>
+	public class CollisionListener : Component, ICmpCollisionListener
+	{
+		public event EventHandler<CollisionEventArgs> CollisionBegin;
```

Comment:
In the same vein, prefer internally storing received events in a list for later validation over redirecting events. Probably need a custom local struct for this, containing event type (local enum) and args.